### PR TITLE
Fix: Multivariate Ridge Regression coefficients

### PR DESCRIPTION
**Pull request by DDJHB**
### Description 
Ridge penalties were passed to daal ridge training in incorrect shape, resulting only in first target's coefs being calculated accurately. This pr addresses the problem by transforming the penalties into correct shape.

Fixes # - [275](https://github.com/intel/scikit-learn-intelex/issues/275)

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

File: daal4py/sklearn/linear_model/tests/test_ridge.py
Comment (ethanglaser) [inline daal4py/sklearn/linear_model/tests/test_ridge.py:1]: it may actually make sense to move this test to sklearnex - we are using the same d4p backend for sklearnex testing but ideally would be able to reuse the test regardless

File: deselected_tests.yaml [hunk 1]
```diff
@@ -208,11 +208,6 @@ deselected_tests:
   # sufficient accuracy (similar to previous cases)
   - linear_model/tests/test_coordinate_descent.py::test_enet_sample_weight_consistency >=0.23
 
-  # The regression coefficients for multi-target problem differ from scikit-learn for small datasets.
-  # Coefficients matches for first label only and the coefficients are close for larger datasets.
-  # See: https://github.com/IntelPython/daal4py/issues/275
-  - linear_model/tests/test_ridge.py::test_ridge_cv_individual_penalties >=0.24
```
Comment (ethanglaser) [inline deselected_tests.yaml:214]: I looked at logs and tbh not sure why this doesn't show up, may need to take a closer look

File: daal4py/sklearn/linear_model/_ridge.py [hunk 2]
```diff
@@ -49,18 +49,20 @@ def _daal4py_fit(self, X, y_):
 
     ridge_params = np.asarray(self.alpha, dtype=X.dtype)
     if ridge_params.size != 1 and ridge_params.size != y.shape[1]:
+        # incorrect order of parameters in the error message is intentional to match sklearn
         raise ValueError(
             "Number of targets and number of penalties do not correspond: "
             f"{ridge_params.size} != {y.shape[1]}"
         )
-    ridge_params = ridge_params.reshape((1, -1))
+    ridge_params = ridge_params.reshape((-1, 1))
```
[the remark this answers is not in the document]
Comment (DDJHB) [reply daal4py/sklearn/linear_model/_ridge.py:57]: not sure about this one, i ve checked and the sklearn tests for multitargets seem to be failing rather for lack of precision on the coef values than just being incorrect. Manipulating shapes did not result into correction.

File: daal4py/sklearn/linear_model/tests/test_ridge.py
Comment (DDJHB) [reply daal4py/sklearn/linear_model/tests/test_ridge.py:1]: i can see your point, but ideally my vision is: this is fixing daal4py (supposedly standalone module) problem, test should exist here and accompany every test run of the module therefore. if it works on this side -> it will work in sklearnex and if it does not -> we know exactly what the problem is. whereas if it does not work on sklearnex you still gotta wonder if that is d4p or sth on sklearnex failing.

File: deselected_tests.yaml (hunk 1 above)
Comment (DDJHB) [reply deselected_tests.yaml:214]: public ci seems to run and pass this test btw

File: daal4py/sklearn/linear_model/tests/test_ridge.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+def test_multivariate_ridge_coefficients():
+    from daal4py.sklearn.linear_model._ridge import Ridge
+
+    X, y = make_regression(n_samples=10, n_features=5, n_targets=3, random_state=0)
+
+    # asserting exception if alpha has wrong shape
+    wrong_alpha_shape = numpy.random.rand(5)
+    with pytest.raises(ValueError):
+        model = Ridge(alpha=wrong_alpha_shape).fit(X, y)
```
Comment (samir-nasibli) [inline daal4py/sklearn/linear_model/tests/test_ridge.py:30]: Make sense to move it into seperate test suit and reuse it on sklearnex side.

File: daal4py/sklearn/linear_model/tests/test_ridge.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy
+import pytest
+from sklearn.datasets import make_regression
+
+
+def test_multivariate_ridge_coefficients():
```
Comment (samir-nasibli) [inline daal4py/sklearn/linear_model/tests/test_ridge.py:22]: I would sugget you to have some common Func:
```python
def _test_multivariate_ridge_coefficients(module, random_state):
   ...


def test_multivariate_ridge_coefficients():
    from daal4py.sklearn.linear_model._ridge import Ridge
    random_state = 0
     _test_multivariate_ridge_coefficients(Ridge, random_state)
``` 

Just import and reuse `_test_multivariate_ridge_coefficients` in tests for sklearnex Ridge #1843 
In `sklearnex/preview/linear_model/tests/test_ridge.py` just add:

```python
def test_multivariate_ridge_coefficients():
    from sklearnex.preview.linear_model import Ridge
    random_state = 0
     _test_multivariate_ridge_coefficients(Ridge, random_state)
```

**Review by icfaust — APPROVED** [review]
I appreciate the new tests and the comment as to the conformance.

**Review by samir-nasibli — APPROVED** [review]
@DDJHB Thank you very much for the work done!
I am very pleased that we have increased test coverage. Looks good to me.
I looked at the fails in internal CI, it seems that they are not related to the PR changes.
CI is considered as green. @ethanglaser please re-check it again

File: sklearnex/linear_model/tests/test_linear.py [hunk 5]
```diff
@@ -124,3 +128,15 @@ def test_sklearnex_reconstruct_model(dataframe, queue, dtype):
...
     assert_allclose(gtr, _as_numpy(y_pred), rtol=tol)
+
+
+def test_sklearnex_multivariate_ridge_coefs():
+    from sklearnex.linear_model import Ridge
+
+    _test_multivariate_ridge_coefficients(Ridge, random_state=0)
+
+
+def test_sklearnex_multivariate_ridge_alpha_shape():
```
Comment (samir-nasibli) [inline sklearnex/linear_model/tests/test_linear.py:139]: Not the scope of this PR, but it make sense as a follow up extend it for the different dataframe inputs check.
